org.metaborg.spoofax.core/src/main/java/org/metaborg/spoofax/core/syntax/JSGLRParseService.java

```diff
@@ -140,204 +89,187 @@ private JSGLRVersion jsglrVersion(ISpoofaxInputUnit input) {
+    @Override public void invalidateCache(ILanguageImpl impl) {
+        if(hasIncrementalPTGen(impl)) {
+            logger.debug("Storing reference parse table for {}", impl);
+            updateReferenceParseTables(impl, false, parserConfigs, referenceParseTables);
+            updateReferenceParseTables(impl, true, completionParserConfigs, referenceCompletionParseTables);
+        }
 
-            FileObject parseTable = null;
-            boolean incrementalPTGen = false;
-            for(ILanguageComponent component : lang.components()) {
-                if(component.config().sdfEnabled()
-                    && component.config().sdf2tableVersion() == Sdf2tableVersion.incremental) {
-                    incrementalPTGen = true;
-                }
-            }
+        logger.debug("Removing cached parse table for {}", impl);
+        parserConfigs.remove(impl);
+        completionParserConfigs.remove(impl);
+        parsers.remove(impl);
+        completionParsers.remove(impl);
+    }
 
-            if(facet.parseTable == null) {
-                try {
-                    boolean multipleTables = false;
-                    for(ILanguageComponent component : lang.components()) {
-                        if(component.config().sdfEnabled()) {
-                            if(component.config().parseTable() != null) {
-                                if(multipleTables) {
-                                    logger.error("Different components are specifying multiple parse tables.");
-                                    throw new ParseException(input);
-                                }
+    @Override public void invalidateCache(ILanguageComponent component) {
 
-                                parseTable = component.location().resolveFile(component.config().parseTable());
-                                multipleTables = true;
-                            }
-                        }
-                    }
-                } catch(FileSystemException e) {
-                    logger.error("Parse table not found or sdf is not enabled for this language.");
-                    throw new ParseException(input, e);
-                }
-            } else {
-                parseTable = facet.parseTable;
-            }
+    }
 
-            try {
-                if(parseTable == null || !parseTable.exists()) {
-                    logger.error("Parse table not found or sdf is not enabled for this language.");
-                    throw new ParseException(input);
-                }
-            } catch(FileSystemException e) {
-                logger.error("Parse table not found or sdf is not enabled for this language.");
-                throw new ParseException(input, e);
-            }
+    @Override public void close() {
+        parserConfigs.clear();
+        completionParserConfigs.clear();
+        referenceParseTables.clear();
+        referenceCompletionParseTables.clear();
```

Comment:
`parsers` and `completionParsers` should probably be cleared here?

Reply:
Yes they should, thanks for the catch! Opened up #43 to fix this :slightly_smiling_face: